src/test/java/org/folio/rest/api/ItemStorageTest.java

```diff
@@ -777,6 +782,47 @@ public void canSearchForItemsByBarcode()
       is("673274826203"));
   }
 
+  @Test
+  public void canSearchForItemsByTags()
+    throws MalformedURLException,
+    InterruptedException,
+    ExecutionException,
+    TimeoutException,
+    UnsupportedEncodingException {
+
+    String tagValue = "important";
+
+    UUID holdingsRecordId = createInstanceAndHolding(mainLibraryLocationId);
+
+    createItem(addTags(tagValue, holdingsRecordId));
+    createItem(nod(holdingsRecordId));
+
+    CompletableFuture<Response> searchCompleted = new CompletableFuture<>();
+
+    String url = itemsStorageUrl("") + "?query=" + URLEncoder.encode("tags.tagList=" + tagValue,
+      StandardCharsets.UTF_8.name());
+
+    client.get(url,
+      StorageTestSuite.TENANT_ID, ResponseHandler.json(searchCompleted));
+
+    Response searchResponse = searchCompleted.get(5, TimeUnit.SECONDS);
+
+    assertThat(searchResponse.getStatusCode(), is(200));
+
+    JsonObject searchBody = searchResponse.getJson();
+
+    JsonArray foundItems = searchBody.getJsonArray("items");
+
+    LinkedHashMap item = (LinkedHashMap) foundItems.getList().get(0);
+    LinkedHashMap<String, ArrayList<String>> itemTags = (LinkedHashMap<String, ArrayList<String>>) item.get("tags");
+
+    assertTrue(searchResponse.getBody().contains(tagValue));
```

Comment:
Could you move asserts for _foundItems_, _searchResponse_ and _searchBody_ under item? to check, for example, items size before you retrieve first element from list

Reply:
Thanks, I'll fix it.